modules/mobile/app/helpers/mobile/pagination_helper.rb

```diff
@@ -56,22 +59,45 @@ def links(number_of_pages)
     end
 
     def link_url(display_page_number)
-      "#{url_without_page_number}&page[number]=#{display_page_number}"
+      params = ["page[size]=#{page_size}", "page[number]=#{display_page_number}", *optional_query_strings].join('&')
+      "#{url}?#{params}"
     end
 
-    def url_without_page_number
-      @url_without_page_number ||= begin
+    # rubocop:disable Metrics/MethodLength
+    def optional_query_strings
+      @optional_query_strings ||= begin
         query_strings = []
-        %w[start_date end_date use_cache show_completed].each do |key|
-          next unless validated_params.key?(key)
 
-          camelized = key.camelize(:lower)
-          query_strings << "#{camelized}=#{validated_params[key]}"
+        optional_params.each_pair do |key, value|
+          next if value.blank?
+
+          camelized = key.to_s.camelize(:lower)
+          case value
+          when Array
+            value.each do |v|
+              query_strings << "#{camelized}[]=#{v}"
+            end
+          when Hash
+            # this is an incomplete implementation that serves our current needs
+            # this will need additional work if we ever need to use more complex hashes
+            first_key = value.keys.first
```

Comment:
This code looks like it can fail if the Hash doesn't contain another hash. I started to do some better enforcement, but this code is only used by one controller which uses a service that fails if the hash is malformed. I decided to just leave this alone for now and we can fix it if problems ever arise.

Reply:
If there isn't logging to catch this specific issue it might be worth adding. Probably in a separate PR

Reply:
Yeah, I'll give this a little more attention. Logging might be the happy medium between pre-optimization and just letting errors happen if they're going to happen.